ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineGetPayloadBodiesByHashV1.java

```diff
@@ -0,0 +1,90 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.engine;
+
+import static org.hyperledger.besu.util.Slf4jLambdaHelper.traceLambda;
+
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.ethereum.ProtocolContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.RpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.JsonRpcRequestContext;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.methods.ExecutionEngineJsonRpcMethod;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcError;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcErrorResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.BlockResultFactory;
+import org.hyperledger.besu.ethereum.api.jsonrpc.internal.results.EngineGetPayloadBodiesResultV1;
+import org.hyperledger.besu.ethereum.chain.Blockchain;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import io.vertx.core.Vertx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EngineGetPayloadBodiesByHashV1 extends ExecutionEngineJsonRpcMethod {
+
+  private static final int MAX_REQUEST_BLOCKS = 1024;
+  private static final Logger LOG = LoggerFactory.getLogger(EngineGetPayloadBodiesByHashV1.class);
+  private final BlockResultFactory blockResultFactory;
+
+  public EngineGetPayloadBodiesByHashV1(
+      final Vertx vertx,
+      final ProtocolContext protocolContext,
+      final BlockResultFactory blockResultFactory,
+      final EngineCallListener engineCallListener) {
+    super(vertx, protocolContext, engineCallListener);
+    this.blockResultFactory = blockResultFactory;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.ENGINE_GET_PAYLOAD_BODIES_BY_HASH_V1.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse syncResponse(final JsonRpcRequestContext request) {
+    engineCallListener.executionEngineCalled();
+
+    final Object reqId = request.getRequest().getId();
+
+    List<Hash> blockHashes =
+        Arrays.stream(request.getRequest().getParams())
+            .map(String::valueOf)
+            .map(Hash::fromHexString)
+            .collect(Collectors.toList());
+
+    traceLambda(LOG, "{} parameters: blockHashes {}", () -> getName(), () -> blockHashes);
+
+    if (blockHashes.size() > getMaxRequestBlocks()) {
+      return new JsonRpcErrorResponse(reqId, JsonRpcError.INVALID_RANGE_REQUEST_TOO_LARGE);
+    }
+
+    final Blockchain blockchain = protocolContext.getBlockchain();
+
+    final EngineGetPayloadBodiesResultV1 engineGetPayloadBodiesResultV1 =
+        blockResultFactory.payloadBodiesCompleteV1(
```

Comment:
It's 'may not' not 'must not'

Reply:
My understanding was the same of what @siladu said. I'm considering that everything that isn't specified with MUST is not a item that we're obliged to comply with. Plus, it's likely to add extra work for Besu to check if the blocks are finalized since we currently only keep track of the latest finalized block